python/GafferUI/EditScopeUI.py

```diff
@@ -380,6 +390,58 @@ def __userNodes( editScope ) :
 		nodes = Gaffer.Metadata.nodesWithMetadata( editScope, "editScope:includeInNavigationMenu" )
 		return [ n for n in nodes if n.ancestor( Gaffer.EditScope ).isSame( editScope ) ]
 
+	def __dropNode( self,  event ) :
+
+		node = self.__inputNode()
+		if node is None :
+			return None
+
+		if isinstance( event.data, Gaffer.EditScope ) :
+			return event.data
+		elif (
+			isinstance( event.data, Gaffer.Set ) and event.data.size() == 1 and
+			isinstance( event.data[0], Gaffer.EditScope )
+		) :
+			return event.data[0]
+		else:
+			return None
+
+	def __dragEnter( self, widget, event ) :
+
+		if self.__dropNode( event ) :
+			self.__frame.setHighlighted( True )
+
+		return True
+
+	def __dragLeave( self, widget, event ) :
+
+		self.__frame.setHighlighted( False )
+
+		return True
+
+	def __drop( self, widget, event ) :
+
+		dropNode = self.__dropNode( event )
+
+		if dropNode :
+			inputNode = self.__inputNode()
+			upstream = Gaffer.NodeAlgo.findAllUpstream( inputNode, self.__editScopePredicate )
+			if self.__editScopePredicate( inputNode ) :
+				upstream.insert( 0, inputNode )
+
+			if dropNode in upstream :
+				self.__connectEditScope( dropNode )
+			else :
+				with GafferUI.PopupWindow() as self.__popup :
```

Comment:
I really like these more ephemeral popup windows for this sort of advisory stuff - feels much less catastrophic and more helpful than a full-blown dialogue. 

This isn't something for this PR, but I think it's worth considering using a PopupWindow in the "we didn't delete render passes because they were created upstream" case. I've witnessed several users interact with the current dialogue and they seem to feel like it's an error rather than guidance. I think just doing the disabling and popping up an ephemeral "N passes were disabled rather than deleted, because they were created upstream" info box might make people feel better about it.

Reply:
Yeah, I quite like them too. They provide context while not breaking your flow as much as a dialogue. Will see how that flies with the Render Pass Editor...